core/src/main/resources/org/apache/spark/ui/static/timeline-view.css

```diff
@@ -24,6 +24,66 @@ div#application-timeline, div#job-timeline {
   margin-top: 5px;
 }
 
+#task-assignment-timeline div.legend-area {
+  border: 1px solid #000000;
```

Comment:
nit: I think this looks better with no border, because I think the border draws the eye too much towards the legend rather than to the visualization (although happy to be overruled if others have strong feelings otherwise)